R/detect_mistakes.R

```diff
@@ -203,15 +204,37 @@ detect_mistakes <- function(user,
       )
     }
 
+    
+    if (length(well_matched > 0)){
+      matched_user_names <- sort(rlang::names2(well_matched))
+      
+    if ( !allow_partial_matching ){
+      ## where does partial matching occur ?
+      where_pmatches <- function(user_name) {
+        which(startsWith(remaining_solution_names, user_name))
```

Comment:
Returning the boolean vector should still work. Since we're not worried about the actual locations, we can remove `which`.
```suggestion
        startsWith(remaining_solution_names, user_name)
```

Reply:
I made some tests and this return an error, 
And we need the location to be sure to create the correct message at the end ( correct argument with correct value)

Reply:
Ok.

We should also check for multiple matches.  Can you add this test case?


```r
# setup
f <- function(ab, abc, abcd) { 1 }
# user
f(a = 1)
```

Reply:
I made a lot of unit test based on this case : 
everything looks fine : 

https://github.com/rstudio-education/gradethis/pull/161/commits/add6afba3c34f714b8e2ba67dfd5785b218f108d